packages/platform-sdk/src/dto/transaction.ts

```diff
@@ -31,7 +31,11 @@ export abstract class AbstractTransactionData {
 		magistrate: "isMagistrate",
 	};
 
-	public constructor(protected readonly data: KeyValuePair) {}
+	protected readonly decimals?: number;
+
+	public constructor(protected readonly data: KeyValuePair, decimals?: number | string) {
+		this.decimals = typeof decimals === "string" ? parseInt(decimals) : decimals;
+	}
```

Comment:
I realize introducing a second argument makes things a little hairy here. Calling code and helpers become more complicated. I may switch this to a setter method for this

Reply:
Think an argument is fine for now. I'm currently reviewing classes like this and thinking about how they could be cleaned up and attach them to a factory.